staging/src/github.com/kubeedge/beehive/pkg/core/module.go

```diff
@@ -18,27 +17,76 @@ type Module interface {
 
 var (
 	// Modules map
-	modules         map[string]Module
-	disabledModules map[string]Module
+	modules         map[string]moduleInfo
+	disabledModules map[string]moduleInfo
 )
 
 func init() {
-	modules = make(map[string]Module)
-	disabledModules = make(map[string]Module)
+	modules = make(map[string]moduleInfo)
+	disabledModules = make(map[string]moduleInfo)
+}
+
+// moduleInfo represent a module info
+type moduleInfo struct {
+	moduleType string
+	remote     bool
+	module     Module
 }
 
 // Register register module
-func Register(m Module) {
-	if m.Enable() {
-		modules[m.Name()] = m
-		klog.Infof("Module %v registered successfully", m.Name())
+// if not passed in parameter opts, default moduleType is "channel"
+func Register(m Module, opts ...string) {
+	var info moduleInfo
+	if len(opts) == 0 {
+		info = moduleInfo{
+			module:     m,
+			moduleType: common.MsgCtxTypeChannel,
+			remote:     false,
+		}
 	} else {
```

Comment:
The else here can be deleted.

Reply:
need to distiguish moduleType by length.